fuzzywuzzy/fuzz.py

```diff
@@ -40,33 +40,22 @@
 # Basic Scoring Functions #
 ###########################
 
+@utils.check_for_none
+@utils.check_empty_string
 def ratio(s1, s2):
-
-    if s1 is None:
-        raise TypeError("s1 is None")
-    if s2 is None:
-        raise TypeError("s2 is None")
     s1, s2 = utils.make_type_consistent(s1, s2)
 
-    if len(s1) == 0 or len(s2) == 0:
```

Comment:
So it looks like here we're doing a type conversion _before_ checking the string length, but your change checks the string length _before_ doing the type conversion. Thoughts?